### ISSUE TITLE
Inference not working for DPO VLM

### ISSUE THREAD

**Issue by schwarzwalder93** [huggingface/trl#3251]
### Reproduction

I have used the exact script here to train Idefics2 using DPO on the dataset mentioned in the script.
https://github.com/huggingface/trl/blob/main/examples/scripts/dpo_vlm.py

Change: I had to set `gradient_checkpointing_kwargs={"use_reentrant": False}` to resolve the training error

The training runs successfully, training/val loss reduces and produces model artefacts.

However, on loading the model for inference, the trained model (with LoRA adapters with non-zero weights) produces same results as the base model, 

```
# Load
model = AutoModelForVision2Seq.from_pretrained("HuggingFaceM4/idefics2-8b", torch_dtype=torch.bfloat16).to("cuda")
processor = AutoProcessor.from_pretrained("HuggingFaceM4/idefics2-8b", do_image_splitting=False)
model.load_adapter('/local/home/trl/dpo_idefics_rlaif-v')

# Process
user_message = 'User:<image>What is the setting or environment in which the image takes place?<end_of_utterance>\n'
image_path = 
images = [Image.open(image_path)]
data = [{"role": "user", "content": [{"type": "image"}, {"type": "text", "text": user_message}]}]
prompts = processor.apply_chat_template(data, add_generation_prompt=True)
inputs = processor(images, prompts, return_tensors="pt")
inputs = {k: v.to("cuda") for k, v in inputs.items()}

# Generate
print('-----Base Model------')
model.disable_adapters()
generated_ids = model.generate(**inputs, max_new_tokens=500)
response_text = processor.batch_decode(generated_ids, skip_special_tokens=True)[0]
print(response_text)

# Generate
print('----DPO Fine-Tuned Model-------')
model.enable_adapters()
generated_ids = model.generate(**inputs, max_new_tokens=500)
response_text = processor.batch_decode(generated_ids, skip_special_tokens=True)[0]
print(response_text)
```
Output:
```
-----Base Model------
User: What is the setting or environment in which the image takes place? 

The image takes place in a forest setting.

----DPO Fine-Tuned Model-------
User: What is the setting or environment in which the image takes place?

The image takes place in a forest setting.
```

### System Info

- Platform: Linux-5.10.228-219.884.amzn2.x86_64-x86_64-with-glibc2.31
- Python version: 3.10.8
- TRL version: 0.16.1
- PyTorch version: 2.6.0
- CUDA device(s): NVIDIA H200, NVIDIA H200, NVIDIA H200, NVIDIA H200, NVIDIA H200, NVIDIA H200, NVIDIA H200, NVIDIA H200
- Transformers version: 4.51.0
- Accelerate version: 1.6.0
- Accelerate config: not found
- Datasets version: 3.5.0
- HF Hub version: 0.30.1
- bitsandbytes version: 0.45.4
- DeepSpeed version: 0.16.5
- Diffusers version: not installed
- Liger-Kernel version: not installed
- LLM-Blender version: not installed
- OpenAI version: not installed
- PEFT version: 0.15.1
- vLLM version: not installed

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

**Comment by schwarzwalder93** [comment]
@qgallouedec @vwxyzjn @kashif FYI

**Comment by ShivMoh** [comment]
hi, I'm also having this issue. Do you know if there are any known fixes or workarounds in the meantime? I'm pretty new to this and not quite sure where to go from here.

@schwarzwalder93 were you able to resolve this in any way?

**Comment by Meiao-W** [comment]
try to change your prompt to this⬇️. To magnify the difference, you should make the output as long as possible.
user_message = 'User:<image>Describe the key features of the image in great detail.<end_of_utterance>\n'

**Comment by qgallouedec** [comment]
DPO is being refactored in #3906 so I'll close this issue, if you think it needs to stay open please let me know
